### PR TITLE
feat!: Allow custom ECR, rename var.use_ecr_image, and cleanup

### LINKED ISSUE #48: Ability to specify custom ECR image
When we have multiple CodeBuild projects spanning multiple repos it can be useful to just have a single ECR repo and image for all those projects especially if they are very similar or if groups of them are similar.

### PR DESCRIPTION
**Pull request by baolsen**
Closes #48 

Breaking change, as I decided to change one of the variables a bit to support more use cases.

The plan is to test this branch at the customer before merging it in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
```
Comment (baolsen) [inline README.md:2]: I moved some sections around to make it easier to read

File: README.md [hunk 2]
```diff
@@ -57,59 +58,22 @@ For example:
 
 # How to use it
```
Comment (baolsen) [inline README.md:60]: Moved the auth steps into their own file

File: README.md [hunk 3]
```diff
@@ -57,59 +58,22 @@ For example:
...
-
-### Adding your token to Parameter Store (Optional)
-
-Add it to AWS Systems Manager Parameter Store with the `SecureString` type.
-
-[![Parameter Store configuration](https://github.com/cloudandthings/terraform-aws-github-runners/blob/main/docs/images/ssm.png)](https://github.com/cloudandthings/terraform-aws-github-runners/blob/main/docs/images/ssm.png )
+## 1. Setup authentication to GitHub
 
+There are several ways to set up authentication to GitHub.
+Follow the guide [here](/docs/GITHUB-AUTH-SETUP.md).
```
Comment (baolsen) [inline README.md:64]: Before merging I probably need to make this a URI like the others. Or it wont work from the TF registry.

File: main.tf [hunk 4]
```diff
@@ -1,18 +1,30 @@
...
-  service_role  = local.create_iam_role ? aws_iam_role.this[0].arn : "arn:aws:iam::${local.aws_account_id}:role/${var.iam_role_name}"
+  service_role = (
+    local.create_iam_role
+    ? aws_iam_role.this[0].arn
+    : "arn:aws:iam::${local.aws_account_id}:role/${var.iam_role_name}"
+  )
 
   artifacts {
     type = "NO_ARTIFACTS"
   }
```
Comment (baolsen) [inline main.tf:18]: Tweaked the logic a bit

**Comment by baolsen** [comment]
I'll also add a precommit hook for codespell for completeness

**Review by andrewchees — APPROVED** [review]

**Pull request merged**